src/js_realm_object.hpp

```diff
@@ -352,6 +354,22 @@ void RealmObjectClass<T>::get_object_key(ContextType ctx, ObjectType object, Arg
     return_value.set(std::to_string(obj_key.value));
 }
 
+template <typename T>
+void RealmObjectClass<T>::get_table_key(ContextType ctx, ObjectType object, Arguments& args,
+                                         ReturnValue& return_value)
+{
+    args.validate_maximum(0);
+
+    auto realm_object = get_internal<T, RealmObjectClass<T>>(ctx, object);
+    if (!realm_object) {
+        throw std::runtime_error("Invalid 'this' object");
+    }
+
+    const Obj& obj = realm_object->obj();
+    auto table_key = obj.get_table()->get_key();
```

Comment:
This seems like something to address in core: there is a private `get_table_key()` method which is just a shortcut to this. I think it'd be good to make that method public since I imagine this would be nice to use in contexts like this (and you can access it anyways so no reason not to).

Reply:
That's an interesting observation, and I suggest that you bring it up with the team working on the bindgen project (core's public API is part of the project).